File: backend/src/main/java/com/sprouthub/sprouthub/service/PostService.java
package com.sprouthub.sprouthub.service;

import com.sprouthub.sprouthub.model.Post;

import java.util.List;

public interface PostService {
    Post createPost(Post post);
    Post getPostById(String id);
    List<Post> getAllPosts();
    Post updatePost(String id, Post post);
    void deletePost(String id);
}

File: frontend/src/services/postService.js
import api from '../utils/api';

const getAllPosts = () => {
  return api.get('/posts');
};

const getPostById = (id) => {
  return api.get(`/posts/${id}`);
};

const createPost = (data) => {
  return api.post('/posts', data);
};

const updatePost = (id, data) => {
  return api.put(`/posts/${id}`, data);
};

const deletePost = (id) => {
  return api.delete(`/posts/${id}`);
};

const postService = {
  getAllPosts,
  getPostById,
  createPost,
  updatePost,
  deletePost,
};

export default postService;